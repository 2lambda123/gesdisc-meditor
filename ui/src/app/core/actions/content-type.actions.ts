import { Action } from '@ngrx/store';
import { ContentType } from '../../models/content-type';

export enum ContentTypesActionTypes {
	SelectContentType = '[ContentType] Select Content Type',
	LoadContentTypes = '[ContentType] Load content types',
	LoadComplete = '[ContentType] Load Complete',
	LoadError = '[ContentType] Load Error',
}

export class SelectContentType implements Action {
	readonly type = ContentTypesActionTypes.SelectContentType;

	constructor(public payload: string) { }
}

export class LoadContentTypes implements Action {
	readonly type = ContentTypesActionTypes.LoadContentTypes;

	constructor() { }
}

export class LoadComplete implements Action {
	readonly type = ContentTypesActionTypes.LoadComplete;

	constructor(public payload: ContentType[]) {}
}

export class LoadError implements Action {
	readonly type = ContentTypesActionTypes.LoadError;

	constructor(public payload: any) {}
}


export type ContentTypesActions =
	| SelectContentType
	| LoadContentTypes
	| LoadComplete
	| LoadError;
