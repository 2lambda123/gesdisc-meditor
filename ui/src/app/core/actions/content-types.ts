import { Action } from '@ngrx/store';
import { ContentType } from '@models/content-type';

export enum ContentTypesActionTypes {
	LoadContentTypes = '[ContentType] Add content types',
	LoadComplete = '[ContentType] Load Complete',
  LoadError = '[ContentType] Load Error',
}

export class LoadContentTypes implements Action {
  readonly type = ContentTypesActionTypes.LoadContentTypes;

  // constructor(public payload: ContentType[]) { }
}

export class LoadComplete implements Action {
  readonly type = ContentTypesActionTypes.LoadComplete;

  constructor(public payload: ContentType[]) {}
}

export class LoadError implements Action {
  readonly type = ContentTypesActionTypes.LoadError;

  constructor(public payload: string) {}
}


export type ContentTypesActions =
	LoadContentTypes
	| LoadComplete
	| LoadError;
