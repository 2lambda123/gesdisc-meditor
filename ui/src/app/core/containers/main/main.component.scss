.app-container {
    background: #f0f0f0;
}

.app-version {
    font-size: 10px;
    position: absolute;
    bottom: 0;
    right: 10px;
}

med-toolbar {
    background: #fff;
    margin-bottom: 20px;
    padding: 0 40px;
    -webkit-box-shadow: 0 3px 5px rgba(57, 63, 72, 0.1);
    -moz-box-shadow: 0 3px 5px rgba(57, 63, 72, 0.1);
    box-shadow: 0 3px 5px rgba(57, 63, 72, 0.1);

    img {
        height: 64px;
    }
}

med-toolbar + div {
    background: #fff;
    margin: 0 auto;
    width: 1200px;
    padding: 15px !important;
}
