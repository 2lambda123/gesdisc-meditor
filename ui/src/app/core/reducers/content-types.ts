import { ContentTypesActionTypes, ContentTypesActions } from '../actions/content-types';
import { createEntityAdapter, EntityAdapter, EntityState } from '@ngrx/entity';
import { ContentType } from '@models/content-type';

export interface State extends EntityState<ContentType> {
	selectedContentTypeId: string | null;
}

export const adapter: EntityAdapter<ContentType> = createEntityAdapter<ContentType>({
  selectId: (contentType: ContentType) => contentType.name,
  sortComparer: false,
});

export const initialState: State = adapter.getInitialState({
	selectedContentTypeId: 'Alerts'
});

export function reducer(
	state: State = initialState,
	action: ContentTypesActions
): State {
	switch (action.type) {
		case ContentTypesActionTypes.LoadComplete:
			return adapter.addMany(action.payload, state);

		case ContentTypesActionTypes.SelectContentType:
			return {
				...state,
				selectedContentTypeId: action.payload
			}

		default:
			return state;
	}
}

