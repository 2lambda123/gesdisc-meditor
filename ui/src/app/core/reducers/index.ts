import { createSelector, createFeatureSelector, ActionReducerMap } from '@ngrx/store';
import * as fromContentTypes from './content-types';

export interface ContentTypesState {
  contentTypes: fromContentTypes.State;
}

export interface State {
  contentTypes: ContentTypesState;
}

export const reducers = {
  contentTypes: fromContentTypes.reducer
};

export const selectContentTypesState = createFeatureSelector<ContentTypesState>('contentTypes');


export const getContentTypesEntitiesState = createSelector(
  selectContentTypesState,
  state => state.contentTypes
);

export const {
  selectIds: getContentTypeIds,
  selectEntities: getContentTypeEntities,
  selectAll: getAllContentTypes,
  selectTotal: getTotalContentTypes,
} = fromContentTypes.adapter.getSelectors(getContentTypesEntitiesState);
// export const selectContentTypesIds = createSelector(selectContentTypesState, fromContentTypes.selectContentTypesIds);
// export const selectContentTypesEntities = createSelector(selectContentTypesState, fromContentTypes.selectContentTypesEntities);
// export const selectAllContentTypes = createSelector(selectContentTypesState, fromContentTypes.selectAllContentTypes);
// export const selectContentTypesTotal = createSelector(selectContentTypesState, fromContentTypes.selectContentTypesTotal);
// export const selectCurrentUserId = createSelector(selectUserState, fromContentTypes.getSelectedUserId);

// export const selectCurrentUser = createSelector(
//   selectUserEntities,
//   selectCurrentUserId,
//   (userEntities, userId) => userEntities[userId]
// );
