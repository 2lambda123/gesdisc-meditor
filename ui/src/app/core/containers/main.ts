import { Component, ChangeDetectionStrategy, OnInit } from '@angular/core';
import { Store, select } from '@ngrx/store';

import * as fromModel from '../../reducers';
import * as Model from '../actions/model.actions';

@Component({
  selector: 'meditor-app',
  changeDetection: ChangeDetectionStrategy.OnPush,
  template: `
    <div fxFill fxLayout="column">
      <med-toolbar fxFlex="60px">
      </med-toolbar>
      <div fxFlex="100">
        <router-outlet></router-outlet>
      </div>
    </div>
  `
  ,
})
export class MainComponent implements OnInit {

  constructor(private store: Store<fromModel.State>) {

  }

  ngOnInit() {
  	this.store.dispatch(new Model.Load());
  }

}
