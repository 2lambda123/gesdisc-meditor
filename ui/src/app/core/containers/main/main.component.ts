import { Component, ChangeDetectionStrategy, OnInit } from '@angular/core';
import { environment } from 'environments/environment';

@Component({
  selector: 'meditor-app',
  changeDetection: ChangeDetectionStrategy.OnPush,
  template: `
    <div class="app-container" fxFill fxLayout="column" fxLayoutAlign="center none">
      <med-toolbar fxFlex="82px"></med-toolbar>
      <div fxFlex="90-100px" style="overflow: scroll; padding: 0 16px;">
        <router-outlet></router-outlet>
      </div>
      <div class="app-version">v{{version}}</div>
      <med-loading></med-loading>
      <med-notification></med-notification>
    </div>
  `,
  styleUrls: ['main.component.scss']
})
export class MainComponent implements OnInit {

  version: string = environment.VERSION

  constructor() {}

	ngOnInit() {}

}
