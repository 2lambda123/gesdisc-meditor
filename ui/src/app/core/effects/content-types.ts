import { Inject, Injectable, InjectionToken, Optional } from '@angular/core';
import { Actions, Effect, ofType } from '@ngrx/effects';
import { Action } from '@ngrx/store';
import { Observable } from 'rxjs';
import {
	catchError,
	debounceTime,
	map,
	skip,
	switchMap,
	takeUntil,
} from 'rxjs/operators';

import { ContentTypeService } from '../services/content-type/content-type.service';
import {
	ContentTypesActionTypes,
	LoadContentTypes,
	LoadComplete,
	LoadError
} from '../actions/content-types';
import { ContentType } from '@models/content-type';

/**
 * Effects offer a way to isolate and easily test side-effects within your
 * application.
 *
 * If you are unfamiliar with the operators being used in these examples, please
 * check out the sources below:
 *
 * Official Docs: http://reactivex.io/rxjs/manual/overview.html#categories-of-operators
 * RxJS 5 Operators By Example: https://gist.github.com/btroncone/d6cf141d6f2c00dc6b35
 */

@Injectable()
export class ContentTypesEffects {
	@Effect()
	load$: Observable<Action> = this.actions$.pipe(
		ofType<LoadContentTypes>(ContentTypesActionTypes.LoadContentTypes),
		switchMap(() => this.contentTypesService.getContentTypes()),
		map((contentTypes: ContentType[]) =>  new LoadComplete(contentTypes)),
		// catchError(err => of(new LoadError(err)))
	);

	constructor(
		private actions$: Actions,
		private contentTypesService: ContentTypeService) {}
		/**
		 * You inject an optional Scheduler that will be undefined
		 * in normal application usage, but its injected here so that you can mock out
		 * during testing using the RxJS TestScheduler for simulating passages of time.
		 */

}
