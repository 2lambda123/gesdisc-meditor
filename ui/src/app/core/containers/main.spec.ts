import { ComponentFixture, TestBed } from '@angular/core/testing';
import { combineReducers, Store, StoreModule } from '@ngrx/store';
import { MainComponent } from './main';
import { NoopAnimationsModule } from '@angular/platform-browser/animations';
import * as fromContentTypes from '../../reducers';
import { NO_ERRORS_SCHEMA } from '@angular/core';
import * as ContentTypes from '../actions/content-types';

describe('Main Page', () => {
	let fixture: ComponentFixture<MainComponent>;
	let store: Store<fromContentTypes.State>;
	let instance: MainComponent;

	beforeEach(() => {
		TestBed.configureTestingModule({
			imports: [
				NoopAnimationsModule,
				StoreModule.forRoot({
					contentTypes: combineReducers(fromContentTypes.reducers),
				})
			],
			declarations: [
				MainComponent
			],
			schemas: [ NO_ERRORS_SCHEMA ]
		});

		fixture = TestBed.createComponent(MainComponent);
		instance = fixture.componentInstance;
		store = TestBed.get(Store);

		spyOn(store, 'dispatch').and.callThrough();
	});

	it('should compile', () => {
		fixture.detectChanges();

		expect(fixture).toMatchSnapshot();
	});

	it('should dispatch a ContentTypes.LoadContentTypes on init', () => {
    const action = new ContentTypes.LoadContentTypes();

    fixture.detectChanges();

    expect(store.dispatch).toHaveBeenCalledWith(action);
  });

});
