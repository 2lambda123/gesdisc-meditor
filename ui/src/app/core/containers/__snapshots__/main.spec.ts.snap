// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main Page should compile 1`] = `
<meditor-app
  store={[Function Store]}
>
  
      <div
    fxfill=""
    fxlayout="column"
  >
    
        
    <med-toolbar
      fxflex="60px"
    >
      
        
    </med-toolbar>
    
        
    <div
      fxflex="100"
    >
      
          
      <router-outlet />
      
        
    </div>
    
      
  </div>
    
</meditor-app>
`;
